src/core.js

```diff
@@ -6,10 +6,20 @@
 */
 /*global define, document, sessionStorage, setTimeout, clearTimeout, ActiveXObject, DOMParser, btoa, atob, module */
 
+import { atob, btoa } from 'abab';
 import MD5 from 'md5';
 import SHA1 from 'sha1';
 import utils from 'utils';
 
+let xmldom;
```

Comment:
Same here. Why can't it moved to `_makeGenerator`? I think the readability is degraded. Does it maybe make sense to put all of this feature detection stuff to it's own polyfill file?

Reply:
Same as `nodeXHR`. :+1: for putting feature detection and conditional imports in a separate module (file), I just have to check because I had a lot of issues with conditional imports + babel + webpack.